onecall.go

```diff
@@ -0,0 +1,176 @@
+	Clouds      int       `json:"clouds"`
+	Visibility  int       `json:"visibility"`
+	WindSpeed   float64   `json:"wind_speed"`
+	WindGust    float64   `json:"wind_gust,omitempty"`
+	WindDeg     float64   `json:"wind_deg"`
+	Pop         float64   `json:"pop"`
+	Rain        Rain      `json:"rain,omitempty"`
+	Snow        Snow      `json:"snow,omitempty"`
+	Weather     []Weather `json:"weather"`
+}
+
+type OneCallDailyData struct {
+	Dt        int         `json:"dt"`
+	Sunrise   int         `json:"sunrise"`
+	Sunset    int         `json:"sunset"`
+	Moonrise  int         `json:"moonrise"`
+	Moonset   int         `json:"moonset"`
+	MoonPhase float64     `json:"moon_phase"`
+	Temp      Temperature `json:"temp"`
+	FeelsLike struct {
+		Day   float64 `json:"day"`
+		Night float64 `json:"night"`
+		Eve   float64 `json:"eve"`
+		Morn  float64 `json:"morn"`
+	} `json:"feels_like"`
+	Pressure  int       `json:"pressure"`
+	Humidity  int       `json:"humidity"`
+	DewPoint  float64   `json:"dew_point"`
+	WindSpeed float64   `json:"wind_speed"`
+	WindGust  float64   `json:"wind_gust,omitempty"`
+	WindDeg   float64   `json:"wind_deg"`
+	Clouds    int       `json:"clouds"`
+	Uvi       float64   `json:"uvi"`
+	Pop       float64   `json:"pop"`
+	Rain      float64   `json:"rain,omitempty"`
+	Snow      float64   `json:"snow,omitempty"`
+	Weather   []Weather `json:"weather"`
+}
+
+type OneCallAlertData struct {
+	SenderName  string   `json:"sender_name"`
+	Event       string   `json:"event"`
+	Start       int      `json:"start"`
+	End         int      `json:"end"`
+	Description string   `json:"description"`
+	Tags        []string `json:"tags"`
+}
+
+// NewCurrent returns a new OneCallData pointer with the supplied parameters
+func NewOneCall(unit, lang, key string, excludes []Exclude, options ...Option) (*OneCallData, error) {
+	unitChoice := strings.ToUpper(unit)
+	langChoice := strings.ToUpper(lang)
+
+	c := &OneCallData{
+		Settings: NewSettings(),
+	}
+
+	if ValidDataUnit(unitChoice) {
+		c.Unit = DataUnits[unitChoice]
+	} else {
+		return nil, errUnitUnavailable
+	}
+
+	if ValidLangCode(langChoice) {
+		c.Lang = langChoice
+	} else {
+		return nil, errLangUnavailable
+	}
+
+	var err error
+	c.Excludes, err = ValidExcludes(excludes)
+	if err != nil {
+		return nil, errExcludesUnavailable
+	}
+
+	c.Key, err = setKey(key)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := setOptions(c.Settings, options); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+// OneCallByCoordinates will provide the onecall weather with the
+// provided location coordinates.
+func (w *OneCallData) OneCallByCoordinates(location *Coordinates) error {
+	response, err := w.client.Get(fmt.Sprintf(fmt.Sprintf(onecallURL, "appid=%s&lat=%f&lon=%f&units=%s&lang=%s&exclude=%s"), w.Key, location.Latitude, location.Longitude, w.Unit, w.Lang, w.Excludes))
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+
+	if err = json.NewDecoder(response.Body).Decode(&w); err != nil {
```

Comment:
```suggestion
	return json.NewDecoder(response.Body).Decode(&w); err != nil {
```

Reply:
Here i copied the function of current.go https://github.com/briandowns/openweathermap/blob/0d6745adadf6da1cb603b97e2b946c6f6f47f7b0/current.go#L105

Reply:
We will update the other refs to be correct in another PR. Let's make this change though.